src/bokeh/models/glyphs.py

```diff
@@ -1208,7 +1208,13 @@ def __init__(self, *args, **kwargs) -> None:
     """)
 
 class Rect(XYGlyph, LineGlyph, FillGlyph, HatchGlyph):
-    ''' Render rectangles.
+    ''' Render rectangles, characterised by x and y position, width, height,
+    and angle of rotation.
+
+    .. warning::
+        ``Rect`` glyphs are not well defined on logarithmic scales. Use
+        :class:`~bokeh.models.Block` or :class:`~bokeh.models.Quad` glyphs
+        instead.
```

Comment:
That's assuming `angle != 0`. Otherwise `Rect` is equivalent to `Block` with a different origin (center vs. a corner).

Reply:
Even with `angle == 0` it isn't well-defined. If the `y` axis is logarithmic, what does the `height` mean? We could assume the same as linear axes and put half the `height` above and half the `height` below the `y`. That would be rendered on log axes as a smaller amount above the `y` and a larger amount below it. This isn't good visually but is 
reasonable mathematically. Alternatively one could opt for displaying half in screen coordinates above the `y` and half below, which is good visually but mathematically dubious.

Logarithmic `Block` works because the `height` is always to one side of the `y`, so both approaches give the same answer.